src/main/java/com/playmonumenta/networkchat/channel/ChannelAnnouncement.java

```diff
@@ -81,73 +80,35 @@ public void setAutoJoin(boolean autoJoin) {
 		mAutoJoin = autoJoin;
 	}
 
-	public static void registerNewChannelCommands(String[] baseCommands, List<Argument<?>> prefixArguments) {
-		List<Argument<?>> arguments;
+	public static void registerNewChannelCommands(LiteralArgument newArg, Argument<String> channelArg) {
+		BooleanArgument autoJoinArg = new BooleanArgument("Auto Join");
+		GreedyStringArgument permissionArg = new GreedyStringArgument("Channel Permission");
 
-		for (String baseCommand : baseCommands) {
-			arguments = new ArrayList<>(prefixArguments);
-			// last element of prefixArguments is channel ID
-			arguments.add(new MultiLiteralArgument(CHANNEL_CLASS_ID));
+		for (String baseCommand : ChatCommand.COMMANDS) {
 			new CommandAPICommand(baseCommand)
-				.withArguments(arguments)
+				.withArguments(newArg)
+				.withArguments(channelArg)
+				.withArguments(new LiteralArgument(CHANNEL_CLASS_ID))
```

Comment:
This may be better off as `new MultiLiteralArgument("Channel Type", CHANNEL_CLASS_ID)`, to give an idea of what this field is.